src/pages/restaurant/components/menu/Menu.js

```diff
@@ -0,0 +1,156 @@
+//Libraries
+import React from 'react';
+import PropTypes from 'prop-types';
+
+//Helpers
+import saveItemToLocalStorage from '../../../../utils/saveItemToLocalStorage';
+import {
+  getUpdatedQuantityPerDishById,
+  getTotalItems,
+} from './helpers/menu.helper';
+import getObjectKeys from '../../../../utils/getObjectKeys';
+import produce from 'immer';
+
+//Images
+import emptyCart from '../../../../images/emptycart.webp';
+
+//Constants
+import {
+  INITIAL_STATE,
+  SAVE_ITEM_TO_LOCAL_STORAGE,
+  ADD_NEW_DISH_TO_CART,
+  INCREMENT_DISH_QUANTITY,
+  DECREMENT_DISH_QUANTITY,
+  EMPTY_CART_DESCRIPTION,
+} from './constants/menu.general';
+
+//Components
+import MenuLinks from './components/menuLinks';
+import Dishes from './components/dishes';
+import Cart from './components/cart';
+import Placeholder from '../../../../commonComponents/placeholder';
+
+//CSS
+import './menu.css';
+
+class Menu extends React.Component {
+  state = INITIAL_STATE;
+  resetCartItems() {
+    this.setState({
+      quantityPerDishById: {},
+    });
+  }
+
+  handleCheckout = () => {
+    const { quantityPerDishById } = this.state;
+    saveItemToLocalStorage(SAVE_ITEM_TO_LOCAL_STORAGE, quantityPerDishById);
+    this.resetCartItems();
+  };
+
+  addNewDishToCart = (dishId) => {
+    const { quantityPerDishById } = this.state;
+    const updatedQuantity = getUpdatedQuantityPerDishById(
+      quantityPerDishById,
+      dishId,
+      ADD_NEW_DISH_TO_CART
+    );
+    this.setState(
+      produce((draft) => {
```

Comment:
inline function, create a separate function for this, structure your code like this
const updatedState = producedFunction();

this.setState({ ... })